features/step_definitions/machine.rb

```diff
@@ -13,19 +13,30 @@
       logger.warn "We will skip this scenario"
       skip_this_scenario
     end
+
+  machine_sets = BushSlicer::MachineSetMachineOpenshiftIo.list(user: admin, project: project("openshift-machine-api"))
+    if machine_sets.length == 0
+      logger.warn "worker machineset does not exist, tests running by copying existing machinesets will fail."
+      logger.warn "We will skip this scenario, even though it is IPI deployment"
+      skip_this_scenario
+    end   
   end
 end
 
 Given(/^I have an UPI deployment and machinesets are enabled$/) do
```

Comment:
Why do we invent a new step here? Where is the code to determine it's a UPI deployment?